internal/message/message.go

```diff
@@ -15,17 +15,17 @@ const AmqpBroker Broker = "amqp"
 // SupportedBrokers defines the list of implemented message publishers.
 var SupportedBrokers = []Broker{AmqpBroker}
 
-// Producer defines the operations required by all message producers.
-type Producer interface {
-	Publish(message interface{}) error
+// Publisher defines the operations required by all message producers.
+type Publisher interface {
```

Comment:
Publisher seemed to make sense as a name, though open to suggestions if this is misleading @sonnysideup

Reply:
Seems fine to me and aligns with the [rabbitmq nomenclature](https://www.rabbitmq.com/publishers.html). 👍